pommerman/network/ion_server/network.py

```diff
@@ -0,0 +1,205 @@
+from re import fullmatch
+from gzip import compress
+from rapidjson import loads, dumps, load
+from json import JSONDecodeError
+from uuid import uuid4
+
+# Global variables
+CONCURRENTLY_LOOKING = {"room": {}, "noroom": []}
+PLAYER_WS = {}
+MATCH_PROCESS = {}
+THREAD_LOCK = False
+
+# Functions
+async def decode_message(message, websocket):
+    """Decode the messages recieved from the clients"""
+    if message["i"] == nic.check.value:
+        await websocket.send(
+            dumps({
+                "i": nic.status_ok.value,
+                "p": len(PLAYER_WS),
+                "m": len(MATCH_PROCESS)
+            }))
+    elif message["i"] == nic.match_act.value:
+        if message["sid"] == MATCH_PROCESS[message["u"]]["id"]:
+            MATCH_PROCESS[message["u"]]["act"][MATCH_PROCESS[message["u"]][
+                "pl"].index(message["id"])] = message["act"]
+            MATCH_PROCESS[message["u"]]["recv"][MATCH_PROCESS[message["u"]][
+                "pl"].index(message["id"])] = True
+    elif message["i"] == nic.replay.value:
+        try:
+            with open(
+                    path.join(
+                        path.join(getcwd(), "matches"),
+                        str(message["p"]) + ".json"), 'r') as f:
+                # Note: Registry expression match comes after as it's an expensive operation as compared to file I/O
+                if fullmatch("^[a-z0-9-]*$", message["p"]) is not None:
+                    f = load(f)
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_ok.value, f]), "utf8")))
+                else:
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+        except:
+            await websocket.send(
+                compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+    elif len(PLAYER_WS) < MAX_PLAYERS and (message["i"] == nic.match.value
+                                           or message["i"] == nic.room.value):
+        uuid = str(uuid4())
+        while uuid in PLAYER_WS:
+            uuid = str(uuid4())
+        PLAYER_WS[uuid] = {"ws": websocket}
+        if message["i"] == nic.match.value:
+            CONCURRENTLY_LOOKING["noroom"].append(uuid)
+            PLAYER_WS[uuid]["l"] = True
+        elif message["i"] == nic.room.value:
+            if message["room"] in CONCURRENTLY_LOOKING["room"]:
+                if len(CONCURRENTLY_LOOKING["room"][message["room"]]) < 4:
+                    CONCURRENTLY_LOOKING["room"][message["room"]].append(uuid)
+            else:
+                CONCURRENTLY_LOOKING["room"][message["room"]] = [uuid]
+            PLAYER_WS[uuid]["l"] = False
+            PLAYER_WS[uuid]["r"] = str(message["room"])
+        await websocket.send(dumps({"i": nic.status_reg.value, "id": uuid, "m": MODE}))
+
+
+async def ws_handler(websocket, pth=None): # pylint: disable=unused-argument
+    """Handle the messages recieved by WebSocket"""
```

Comment:
period. 
Also, what are the args?

Reply:
^^

Reply:
-_-

Reply:
Why do you need `pth`?

Reply:
The websocket library returns it but I don't need it and if I don't set it as an argument then the websocket library errors out due to feeding more arguments than necessary.

Reply:
Add that to the doc block so there is no confusion if someone else works on it.